client/senddecorator/metrics.go

```diff
@@ -0,0 +1,53 @@
+package senddecorator
+
+import (
+	"fmt"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/Azure/go-autorest/autorest"
+	"github.com/prometheus/client_golang/prometheus"
+
+	"github.com/giantswarm/azure-operator/v5/service/collector"
+)
+
+const metricsNamespace = "azure_api"
+
+func MetricsDecorator(name, subscriptionID string, metricsCollector collector.AzureAPIMetrics) autorest.SendDecorator {
+	lowerName := strings.ToLower(name)
+
+	// XXX: Following is slightly ugly as code, but better for UX. I buy the tradeoff. - @tuommaki
+	titleName := strings.ReplaceAll(lowerName, "_", " ")
+	titleName = strings.Title(titleName)
+	titleName = strings.ReplaceAll(titleName, "Dns", "DNS")
+	titleName = strings.ReplaceAll(titleName, "Gw", "GW")
+	titleName = strings.ReplaceAll(titleName, "Ip", "IP")
+	titleName = strings.ReplaceAll(titleName, "Nat", "NAT")
+	titleName = strings.ReplaceAll(titleName, "Skus", "SKUs")
+	titleName = strings.ReplaceAll(titleName, "Vms", "VMs")
+
+	totalCallsOpts := prometheus.Opts{Namespace: metricsNamespace, Name: fmt.Sprintf("%s_total_calls", lowerName), Help: fmt.Sprintf("Total number of %s API calls", titleName), ConstLabels: prometheus.Labels{"subscription_id": subscriptionID}}
```

Comment:
Would it be better or worse to have one single metric per type and the lowerName as a label?

Reply:
That's a good shout. I need to think about it how to get it done, because the current prom interfaces don't have `With()` method to supply additional labels. I'll dig around a bit.